configure.ac

```diff
@@ -2014,15 +2014,42 @@ AC_ARG_WITH([maxq10xx],
     ]
 )
 
+AC_ARG_ENABLE([microchip],
+    [AS_HELP_STRING([--enable-microchip],[Enable wolfSSL support for microchip/atmel 508/608/100 (default: disabled)])],
+    [ ENABLED_ATMEL=$enableval ],
+    [ ENABLED_ATMEL=no ]
+    )
+
+if test "$ENABLED_ATMEL" != "no"
+then
+#    AM_CFLAGS="$AM_CFLAGS -DWOLFSSL_ATMEL"
+
+    for v in `echo $ENABLED_ATMEL | tr "," " "`
+    do
+      case $v in
+        508)
+            AM_CFLAGS="$AM_CFLAGS -DWOLFSSL_ATECC508A"
+            ;;
+
+        608)
+            AM_CFLAGS="$AM_CFLAGS -DWOLFSSL_ATECC608A"
+            ;;
+
+        100)
+            AM_CFLAGS="$AM_CFLAGS -DWOLFSSL_ATECC508A -DWOLFSSL_MICROCHIP_TA100 -DMICROCHIP_DEV_TYPE=TA100"
```

Comment:
Why does this include `WOLFSSL_ATECC508A`?

Reply:
Some configurations and codes are common between TA100 and ATECC508A.

Reply:
I'd prefer the sections of code needed be wrapped in WOLFSSL_MICROCHIP_TA100 and not require the ATECC.

Reply:
Still outstanding, simplifying the defines needed helps with user_settings.h builds. I second not requiring WOLFSSL_ATECC508A for type 100.